analysis_options.yaml

```diff
@@ -5,13 +5,169 @@
 # IDEs (https://dart.dev/tools#ides-and-editors). The analyzer can also be
 # invoked from the command line by running `flutter analyze`.
 
-# The following line activates a set of recommended lints for Flutter apps,
-# packages, and plugins designed to encourage good coding practices.
-include: package:flutter_lints/flutter.yaml
-
 linter:
   rules:
     - always_declare_return_types
+    - always_require_non_null_named_parameters
     - always_specify_types
+    - annotate_overrides
+    - avoid_bool_literals_in_conditional_expressions
+    # - avoid_classes_with_only_static_members # ENABLE IN THE FUTURE
+    - avoid_double_and_int_checks
     - avoid_dynamic_calls
-    - prefer_relative_imports
\ No newline at end of file
+    - avoid_empty_else
+    - avoid_escaping_inner_quotes
+    - avoid_field_initializers_in_const_classes
+    - avoid_function_literals_in_foreach_calls
+    - avoid_implementing_value_types
+    - avoid_init_to_null
+    - avoid_js_rounded_ints
+    - avoid_multiple_declarations_per_line
+    - avoid_positional_boolean_parameters
+    - avoid_print
+    - avoid_null_checks_in_equality_operators
+    - avoid_redundant_argument_values
+    - avoid_renaming_method_parameters
+    - avoid_return_types_on_setters
+    - avoid_returning_null
+    - avoid_returning_null_for_future
+    - avoid_returning_null_for_void
+    - avoid_setters_without_getters
+    - avoid_shadowing_type_parameters
+    - avoid_single_cascade_in_expression_statements
+    - avoid_slow_async_io
+    - avoid_type_to_string
+    - avoid_types_as_parameter_names
+    - avoid_unnecessary_containers
+    - avoid_unused_constructor_parameters
+    - avoid_void_async
+    - avoid_web_libraries_in_flutter
+    - await_only_futures
+    - camel_case_extensions
+    - camel_case_types
+    - cancel_subscriptions
+    - cast_nullable_to_non_nullable
+    - conditional_uri_does_not_exist
+    - constant_identifier_names
+    - control_flow_in_finally
+    - curly_braces_in_flow_control_structures
+    - depend_on_referenced_packages
+    - deprecated_consistency
+    - directives_ordering
+    - empty_catches
+    - empty_constructor_bodies
+    - empty_statements
+    - eol_at_end_of_file
+    - exhaustive_cases
+    - file_names
+    - hash_and_equals
+    - implementation_imports
+    - iterable_contains_unrelated_type
+    - leading_newlines_in_multiline_strings
+    - library_names
+    - library_prefixes
+    - library_private_types_in_public_api
+    - list_remove_unrelated_type
+    - missing_whitespace_between_adjacent_strings
+    - no_adjacent_strings_in_list
+    # - no_default_cases # DO WE WANT THIS?
```

Comment:
I think defaults are usefull sometimes but Im ok with both

Reply:
I don't want this. I think this would be more on the functionality and less on the maintainability side because it has the potential to make a lot of code unreadable. In our case, `isMutuallyExclusive` would be problematic with this rule

Reply:
Yes that was my thought. But in that case we could of course put "ignore: no_default_cases" there. But I agree that it can be a bit annoying. We should still try to follow the rule whenever possible, but I can remove it from the list